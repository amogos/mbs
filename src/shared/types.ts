export interface UserRecordType {
    id: number;
    name: string | undefined;
    email: string;
    picture: string;
    following: number[];
    rating: number;
}

export interface UserValueType {
    name: string | undefined;
    email: string;
    picture: string;
    rating: number;
}

export const NullUser: UserRecordType = {
    id: 0,
    name: '',
    email: '',
    picture: 'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
    following: [],
    rating: 0,
};

export const NullLanguage: LanguageRecordType = { id: 0, title: '' };

export interface CategoryRecordType {
    id: number;
    title: string;
}

export interface BookValueType {
    title: string;
    subtitle: string;
    author: string[];
    language: LanguageRecordType;
    image: string;
    owner: UserRecordType;
    holder: UserRecordType;
    state: string;
    isbn: string;
    isbn10: string;
    isbn13: string;
    category: CategoryRecordType;
    return?: number;
    contentScore?: number;
    numReviews?: number;
    format: number;
    space: number;
    description: string;
    length: number;
}

export interface BookDescriptionValueType {
    title: string;
    subtitle: string;
    language: LanguageRecordType;
    image: string;
    author: string[];
    isbn10: string;
    isbn13: string;
    description: string;
    category: number;
    length: number;
    format: number;
}

export interface BookDescriptionRecordType {
    id: number;
    title: string;
    subtitle: string;
    language: LanguageRecordType;
    image: string;
    author: string[];
    isbn10: string;
    isbn13: string;
    description: string;
    length: number;
    format: number;
    category: number;
}

export const NullBookDescriptionRecordType: BookDescriptionRecordType = {
    id: 0,
    title: '',
    subtitle: '',
    language: NullLanguage,
    image: '',
    author: [],
    isbn10: '',
    isbn13: '',
    description: '',
    length: 0,
    format: 0,
    category: 0,
};

export interface BookRecordType {
    id: number;
    title: string;
    subtitle: string;
    author: string[];
    language: LanguageRecordType;
    image: string;
    owner: UserRecordType;
    holder: UserRecordType;
    state: string;
    category: CategoryRecordType;
    format: string;
    space: SpaceType;
    isbn10: string;
    isbn13: string;
    return?: number;
    contentScore?: number;
    numReviews?: number;
    description: string;
}

export interface BookRawValueType {
    isbn10: string;
    isbn13: string;
    owner: number;
    holder: number;
    state: string;
    space: number;
    return?: number;
}

export interface BookRawRecordType {
    id: number;
    isbn10: string;
    isbn13: string;
    owner: number;
    holder: number;
    state: string;
    space: number;
    return?: number;
}

export const NullRawBookRecordType: BookRawRecordType = {
    id: 0,
    owner: 0,
    holder: 0,
    state: '',
    space: 0,
    isbn10: '',
    isbn13: '',
};

export interface LanguageRecordType {
    id: number;
    title: string;
}

export interface QueueNotificationRecordType {
    id: number;
    user: UserRecordType;
    bookTitle: string;
    bookId: number;
    duration: number;
}

export interface QueueNotificationType {
    user: UserRecordType;
    bookTitle: string;
    bookId: number;
    duration: number;
}

export interface ReturnRecordType {
    id: number;
    bookId: number;
    userId: number;
    ownerId: number;
}

export interface ReturnValueType {
    bookId: number;
    userId: number;
    ownerId: number;
}

export interface ReturnNotificationType {
    returnId: number;
    bookId: number;
    bookTitle: string;
    user: UserRecordType;
}

export interface QueueValueType {
    bookId: number;
    userId: number;
    ownerId: number;
    duration: number;
}

export interface QueueRecordType {
    id: number;
    bookId: number;
    userId: number;
    ownerId: number;
    duration: number;
}

export const NullCategory: CategoryRecordType = { id: 0, title: '' };

export interface SpaceType {
    id: number;
    user: UserRecordType;
    numberOfBooks: number;
    numberOfFollowers: number;
    rating: number; // 1-5 and fractional
    transport: number; // 0 - no transport  1 - optional/not mendatory  2 - transport covered
    title: string;
    description: string;
    format: string[];
    picture: string;
}

export const NullSpace: SpaceType = {
    id: 0,
    user: NullUser,
    numberOfBooks: 0,
    numberOfFollowers: 0,
    rating: 0,
    transport: 0,
    title: '',
    description: '',
    format: [],
    picture: '',
};

export interface SpaceRawRecordType {
    id: number;
    owner: number;
    subscription: number;
    title: string;
    description: string;
    transport: number;
    picture: string;
}

export interface BookReviewRecordType {
    id: number;
    bookId: number;
    comment: string;
    contentScore: number;
    stateScore: number;
}

export interface BookReviewValueType {
    bookId: number;
    comment: string;
    contentScore: number;
    stateScore: number;
}

export interface FormatRawType {
    id: number;
    type: string;
}

export const NullFormat: FormatRawType = { id: 0, type: '' };

export interface BannerAttributes {
    showCategoryFiltering: boolean;
    showTabFiltering: boolean;
}

export interface Spaces {
    userSpaces: SpaceType[];
    otherSpaces: SpaceType[];
}

export interface UrlParms {
    id: string;
    query: { category?: number; space?: number };
}

export interface UserReviewRecordType {
    bookId: number;
    userId: number;
    comment: string;
    stateScore: number;
    id: number;
}
