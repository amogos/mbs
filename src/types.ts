export interface UserType {
    name: string;
    email: string;
}

export const IdleBookState = 'state.book.idle';

export interface BookValueType {
    title: string;
    author: string;
    language: string;
    image: string;
    owner: UserType;
    state: string;
    pending: UserType[];
}

export interface BookKeyType {
    id: string | null;
}

export interface BookRecordType {
    id: string | null;
    value: BookValueType;
}

export const nullUser: UserType = { name: '', email: '' };
