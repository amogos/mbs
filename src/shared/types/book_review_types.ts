import { UserRecordType, NullUserRecordType } from './user_types';

export interface BookReviewRecordType {
    id: number;
    bookId: number;
    isbn10: string;
    isbn13: string;
    comment: string;
    score: number;
    user: UserRecordType;
    date: number;
    likes: number;
}

export const NullBookReviewRecordType = (): BookReviewRecordType => {
    return {
        id: 0,
        bookId: 0,
        isbn10: '',
        isbn13: '',
        comment: '',
        score: 0,
        user: NullUserRecordType(),
        date: 0,
        likes: 0,
    };
};

export interface BookReviewRawRecordType {
    id: number;
    bookId: number;
    isbn10: string;
    isbn13: string;
    comment: string;
    score: number;
    user: number;
    date: number;
    likes: number;
}

export const NullBookReviewRawRecordType = (): BookReviewRawRecordType => {
    return { id: 0, bookId: 0, isbn10: '', isbn13: '', comment: '', score: 0, user: 0, date: 0, likes: 0 };
};

export interface BookReviewRawValueType {
    bookId: number;
    isbn10: string;
    isbn13: string;
    comment: string;
    score: number;
    user: number;
    date: number;
    likes: number;
}

export const NullBookReviewRawValueType = (): BookReviewRawValueType => {
    return { bookId: 0, isbn10: '', isbn13: '', comment: '', score: 0, user: 0, date: 0, likes: 0 };
};

export interface BookReviewStatisticsType {
    contentScore: number;
    numReviews: number;
}

export const NullBookReviewStatisticsType = (): BookReviewStatisticsType => {
    return { contentScore: 0, numReviews: 0 };
};

export const ToBookReviewRawRecordType = (from: BookReviewRecordType) => {
    let to: BookReviewRawRecordType = NullBookReviewRawRecordType();
    to = { ...from, user: from.user.id };
    return to;
};
