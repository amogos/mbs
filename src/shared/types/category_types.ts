export interface CategoryRecordType {
    id: number;
    title: string;
}
export const NullCategoryRecordType: CategoryRecordType = { id: 0, title: '' };
