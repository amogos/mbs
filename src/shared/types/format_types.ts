export interface FormatRawType {
    id: number;
    type: string;
}

export const NullFormat: FormatRawType = { id: 0, type: '' };
