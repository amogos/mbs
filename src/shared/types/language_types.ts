export const NullLanguage = (): LanguageRecordType => {
    return { id: 0, title: '' };
};

export interface LanguageRecordType {
    id: number;
    title: string;
}
