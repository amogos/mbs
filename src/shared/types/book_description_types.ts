import { LanguageRecordType, NullLanguage } from './language_types';
import { CategoryRecordType } from './category_types';

export interface BookDescriptionValueType {
    title: string;
    subtitle: string;
    language: LanguageRecordType;
    image: string;
    author: string[];
    isbn10: string;
    isbn13: string;
    description: string;
    category: CategoryRecordType[];
    length: number;
    format: number;
}

export const NullBookDescriptionValueType: BookDescriptionValueType = {
    title: '',
    subtitle: '',
    language: NullLanguage,
    image: '',
    author: [],
    isbn10: '',
    isbn13: '',
    description: '',
    length: 0,
    format: 0,
    category: [],
};

export interface BookDescriptionRecordType {
    id: number;
    title: string;
    subtitle: string;
    language: LanguageRecordType;
    image: string;
    author: string[];
    isbn10: string;
    isbn13: string;
    description: string;
    length: number;
    format: number;
    category: CategoryRecordType[];
}

export const NullBookDescriptionRecordType: BookDescriptionRecordType = {
    id: 0,
    title: '',
    subtitle: '',
    language: NullLanguage,
    image: '',
    author: [],
    isbn10: '',
    isbn13: '',
    description: '',
    length: 0,
    format: 0,
    category: [],
};
