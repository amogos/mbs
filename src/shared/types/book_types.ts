import { LanguageRecordType, NullLanguage } from './language_types';
import { CategoryRecordType, NullCategoryRecordType } from './category_types';
import { UserRecordType, NullUserRecordType } from './user_types';
import { SpaceType, NullSpaceType } from './space_types';
import * as BookStates from './../constants/book_states_constant';

export interface BookValueType {
    title: string;
    subtitle: string;
    author: string[];
    language: LanguageRecordType;
    image: string;
    owner: UserRecordType;
    holder: UserRecordType;
    state: string;
    isbn: string;
    isbn10: string;
    isbn13: string;
    category: CategoryRecordType;
    return?: number;
    contentScore?: number;
    numReviews?: number;
    format: number;
    space: number;
    description: string;
    length: number;
}

export const EmptyBookValueType = () => {
    return {
        title: '',
        subtitle: '',
        author: [],
        language: NullLanguage,
        image:
            'https://vignette.wikia.nocookie.net/superfriends/images/a/a5/No_Photo_Available.jpg/revision/latest?cb=20090329133959',
        owner: NullUserRecordType,
        state: BookStates.default.STATE_BOOK_IDLE,
        isbn: '',
        isbn10: '',
        isbn13: '',
        holder: NullUserRecordType,
        category: NullCategoryRecordType,
        format: 1,
        space: 0,
        description: '',
        length: 0,
    };
};

export interface BookRecordType {
    id: number;
    title: string;
    subtitle: string;
    author: string[];
    language: LanguageRecordType;
    image: string;
    owner: UserRecordType;
    holder: UserRecordType;
    state: string;
    category: CategoryRecordType;
    format: string;
    space: SpaceType;
    isbn10: string;
    isbn13: string;
    return?: number;
    contentScore?: number;
    numReviews?: number;
    description: string;
    length: number;
}

export const NullBookRecordType: BookRecordType = {
    id: 0,
    title: '',
    subtitle: '',
    author: [],
    language: NullLanguage,
    image: '',
    owner: NullUserRecordType,
    holder: NullUserRecordType,
    state: '',
    category: NullCategoryRecordType,
    format: '',
    space: NullSpaceType,
    isbn10: '',
    isbn13: '',
    return: 0,
    contentScore: 0,
    numReviews: 0,
    description: '',
    length: 0,
};

export interface BookRawValueType {
    isbn10: string;
    isbn13: string;
    owner: number;
    holder: number;
    state: string;
    space: number;
    category: number;
    return?: number;
}

export interface BookRawRecordType {
    id: number;
    isbn10: string;
    isbn13: string;
    owner: number;
    holder: number;
    state: string;
    space: number;
    category: number;
    return?: number;
}

export const NullRawBookRecordType: BookRawRecordType = {
    id: 0,
    owner: 0,
    holder: 0,
    state: '',
    space: 0,
    category: 0,
    isbn10: '',
    isbn13: '',
};
