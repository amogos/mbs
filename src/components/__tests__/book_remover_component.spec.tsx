import React from 'react';
import renderer from 'react-test-renderer';
import * as DataTypes from '../../types';
import BookRemoverComponent from './../book_remover_component';

describe('Should render correctly in both cases of ownership', () => {
    test('I am the owner so I should see the delete button', () => {
        const userdata: DataTypes.UserType = { name: 'Iulia Mogos', email: 'daosmistique@yahoo.com', id: 1 };
        const bookValue: DataTypes.BookValueType = {
            author: 'Eric Carle',
            state: 'state.book.idle',
            image: 'https://images-na.ssl-images-amazon.com/images/I/51lsugWtCvL._SY498_BO1,204,203,200_.jpg',
            language: DataTypes.nullLanguage,
            owner: {
                email: 'daosmistique@yahoo.com',
                name: 'Iulia Mogos',
                id: 1,
            },
            holder: DataTypes.nullUser,
            title: "The Very Hungry Caterpillar's ABC",
        };
        const bookKey = 1;
        const props = {
            id: bookKey,
            userdata: userdata,
            value: bookValue,
            deleteBook: jest.fn(),
        };
        const component = renderer.create(<BookRemoverComponent {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('I am not the owner I should not see the delete button ', () => {
        const userdata: DataTypes.UserType = { name: 'PixyDixi', email: 'pixydixy@yahoo.com', id: 1 };
        const bookValue: DataTypes.BookValueType = {
            author: 'Eric Carle',
            state: 'state.book.idle',
            holder: DataTypes.nullUser,
            image: 'https://images-na.ssl-images-amazon.com/images/I/51lsugWtCvL._SY498_BO1,204,203,200_.jpg',
            language: DataTypes.nullLanguage,
            owner: {
                email: 'daosmistique@yahoo.com',
                name: 'Iulia Mogos',
                id: 1,
            },
            title: "The Very Hungry Caterpillar's ABC",
        };
        const bookKey = 1;
        const props = {
            id: bookKey,
            userdata: userdata,
            value: bookValue,
            deleteBook: jest.fn(),
        };
        const component = renderer.create(<BookRemoverComponent {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
