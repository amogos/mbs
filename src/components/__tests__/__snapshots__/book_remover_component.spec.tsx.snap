// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly in both cases of ownership I am not the owner I should not see the delete button  1`] = `null`;

exports[`Should render correctly in both cases of ownership I am the owner so I should see the delete button 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
          Object {
            "color": "#000000ff",
          },
        ]
      }
    >
      delete
    </Text>
  </View>
</View>
`;
